src/notifications/controller.js

```diff
@@ -1,100 +1,80 @@
-const NotificationModel = require('./model.js');
+class NotificationController {
+  db;
 
-const self = this;
+  /**
+   * @constructor
+   * @param {NotificationDatabaseContract} notificationDatabase
+   */
+  constructor(notificationDatabase) {
+    this.db = notificationDatabase;
+  }
 
-/**
- * Method for exposed endpoint. Get all notifications by User id.
- * GET <BASE API>/all/:userId
- * @pathParam {userid} Notification Id
- */
-exports.getAll = (req, res) => {
-  NotificationModel
-  .find({ to: req.params.userid },
-    (err, data) => {
-      if (err) res.send(err);
-      res.send(data);
-    });
-};
+  /**
+   * Creates a new notification.
+   * @param  {Object} data, object representing a notification.
+   */
+  create(data) {
+    return this.db.create(data);
+  }
 
-/**
- * Method for exposed endpoint. Update unread field to false
- * PUT <BASE API>/:notificationId
- * @pathParam {notificationId} Notification Id
- */
-exports.updateUnread = (req, res) => {
-  const id = req.params.notificationId;
-  self
-  .update(id, { unread: false },
-    (err, data) => {
-      if (err) res.send(err);
-      res.send(data);
-    });
-};
+  /**
+   * Updates a notification by id.
+   * @param  {String} id
+   * @param  {Object} data, values to update
+   */
+  update(id, data) {
+    return this.db.update(id, data);
+  }
 
-/**
- * Method for exposed endpoint. Update all unread notifications.
- * PUT <BASE API>/all
- */
-exports.updateAllUnread = (req, res) => {
-  NotificationModel
-  .update({ unread: true }, { unread: false }, { multi: true },
-    (err, data) => {
-      if (err) res.send(err);
-      res.send(data);
-    });
-};
+  /**
+   * Get all notifications by User id.
+   * @param {String} userId
+   */
+  findAllByUser(userId) {
+    return this.db.findAllByUser(userId);
+  }
 
-/**
- * Method for exposed endpoint. Remove notificaton by id
- * DELETE <BASE API>/:notificationId
- * @pathParam {notificationId} Notification Id
- */
-exports.delete = (req, res) => {
-  const id = req.params.notificationId;
-  NotificationModel
-  .findByIdAndRemove(id,
-    (err, data) => {
-      if (err) res.send(err);
-      res.send(data);
-    });
-};
+  /**
+   * Sets the unread field to false, meaning that the
+   * notification is read.
+   * @param {String} id
+   */
+  setAsRead(id) {
+    return this.db.setAsRead(id);
+  }
 
-/**
- * Get unread notification by User id. All the notifications unread.
- * @param  {String}   userId User id
- * @param  {Function} cb     Callback
- */
-exports.getUnread = (userId, cb) => {
-  NotificationModel.find({ to: userId, unread: true }, cb);
-};
+  /**
+   * Sets all the unread notifications to be as 'read' status,
+   * Basically, sets the unread field to be false.
```

Comment:
This sounds awkward, maybe we could change this text to "Sets all unread notifications as read, basically sets the unread field to false"